### PR TITLE
Update license

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR updates the license to reflect the move to the `dask` org

cc @jcrist does this seem reasonable to you?

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2016, Jim Crist
+Copyright (c) 2016-2022, Jim Crist. 2023, Dask Developers.
```
Comment (jcrist) [inline LICENSE:1]: Updating last name. Also, AFAIK you should only have to put the start year on the license.

```suggestion
Copyright (c) 2016, Jim Crist-Harif. 2023, Dask Developers.
```

**Review by jcrist — APPROVED** [review]
Small nit, otherwise LGTM!

File: LICENSE (hunk 1 above)
Comment (jrbourbeau) [reply LICENSE]: Of course. A lots changed since 2016 : )

**Comment by jrbourbeau** [comment]
Just noting that the pre-commit failure is unrelated to the changes here. I'm going to merge after other CI builds are done and handle pre-commit separately.

**Pull request merged**